sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -178,7 +178,12 @@ private[sql] abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
 
       // Pick CartesianProduct for InnerJoin
       case logical.Join(left, right, Inner, condition) =>
-        joins.CartesianProductExec(planLater(left), planLater(right), condition) :: Nil
+        if (conf.cartesianProductEnabled) {
+          joins.CartesianProductExec(planLater(left), planLater(right), condition) :: Nil
+        } else {
+          throw new AnalysisException("Cartesian products are disabled by default. To explicitly " +
```

Comment:
Cartesian joins? Might be confusing for some users to hear cartesian product. If you say join they will know it's the join.